### PR TITLE
feat(helm): add --with-subcharts option for "helm inspect values"

### LINKED ISSUE #3926: Proposal: --merge option for "helm inspect values"
## Motivation

Running `helm inspect values [CHART]` outputs the contents of a chart's `values.yaml` verbatim. This is often used to determine which configuration options a chart exposes. However, it does not provide the contents of `values.yaml` files of potential subcharts bundled in the `charts` subdirectory.

## Proposal

I would like to propose a new command-line option `--merge` for the `helm inspect values` sub-command.

This would cause Helm to merge the `values.yaml` files of a chart and all its subcharts and output the result. The code would need to handle prefixing value names from subcharts with the respective chart names, except for `global` values.

## Sample

Contents of `mychart/values.yaml`:
```yaml
somevar: 123
global:
  sample1: 'abc'
  sample2: 'def'
```

Contents of `mychart/charts/subchart/values.yaml`:
```yaml
somevar: 456
global:
  sample2: 'xyz'
  sample3: 'xyz'
```

Output of `helm inspect values --merge mychart`:
```yaml
somevar: 123 # from mychart
subchart: # prefixed with subchart name
  somevar: 456 # from subchart
global: # no prefixing for global
  sample1: 'abc' # from mychart
  sample2: 'def' # from mychart, value from subchart ignored
  sample3: 'xyz' # from subchart
```

## Implementation

If you guys think this is a sensible approach we (@andreasbieber, @typusomega, @bastianeicher) would be happy to give a PR a try.

### PR DESCRIPTION
**Pull request by AndreasBieber**
Running helm inspect values [CHART] outputs the contents of a chart's values.yaml verbatim. This is often used to determine which configuration options a chart exposes. However, it does not provide the contents of values.yaml files of potential subcharts bundled in the charts subdirectory. The new command-line option --with-subcharts causes Helm to merge the values.yaml files of a chart and all its subcharts and output the result.

Closes #3926

### CODE REVIEW

**Comment by AndreasBieber** [comment]
Signing CLA is done

**Comment by AndreasBieber** [comment]
@k8s-bot /check-cla

**Comment by AndreasBieber** [comment]
@k8s-ci-robot check-cla

**Comment by AndreasBieber** [comment]
Signing CLA done

**Comment by AndreasBieber** [comment]
Signing CLA done

**Comment by bastianeicher** [comment]
Hi guys,
just wanted to check-in to see it anything is missing for this to be merged.
Regards
Bastian

**Comment by irajdeep** [comment]
Hi guys.
Can this be merged looks , it's a super useful feature

**Comment by freach** [comment]
What's the status here, this is a very useful feature.

**Comment by carpenterbees** [comment]
Also waiting for this feature to be merged.

**Comment by dennisosimon** [comment]
this would be really helpful!

File: cmd/helm/inspect_test.go [hunk 1]
```diff
@@ -78,3 +78,20 @@ func TestInspect(t *testing.T) {
...
+	expected, _ := ioutil.ReadFile("testdata/testcharts/chart-globals-inspect-values-with-subcharts/expected.yml")
+	b := bytes.NewBuffer(nil)
+
+	insp := &inspectCmd{
+		chartpath:     "testdata/testcharts/chart-globals-inspect-values-with-subcharts",
+		output:        all,
+		out:           b,
+		withSubcharts: true,
+	}
+	insp.run()
```
Comment (bacongobbler) [inline cmd/helm/inspect_test.go]: no error checking?

**Review by bacongobbler — CHANGES_REQUESTED** [review]
Hey @AndreasBieber, just letting you know that as per [the announcement on the mailing list](https://lists.cncf.io/g/cncf-helm/topic/last_v2_feature_release/34301238), Helm 2.15.0 is the last release where we will be merging new features for Helm 2. If you want to get this in for the next release, please let us know and we can work with you to get this in. If not, we'll have to close this and continue this effort over in Helm 3.

I was very confused after reading the expected testdata output. I was having a hard time looking at the testdata output, ensuring that the values are displayed as expeected. Values with names like `chart-globals-inspect-values-with-subcharts-subchart1-subsubchart` is very verbose and a little hard to understand. Is there a simpler naming convention you can come up with here to make the code a little simpler to understand?

Thanks!

**Comment by bacongobbler** [comment]
I'm going to close this ticket due to inactivity, but please re-open if this still needs to be addressed for Helm 3. Thanks!

**Pull request closed without merging**